state_machine/drivers/pycubedmini/lib/pycubed.py

```diff
@@ -262,394 +306,168 @@ def reinit(self, dev):
-
-        # open the current file directory as read only, print line by line
-        # (removing whitespace)
-        with open(filedir, "r") as file:
-            for line in file:
-                print(line.strip())
-
-    def send_file(self, c_size, send_buffer, filename):
-        """
-        send a file given character size, buffer size, and the filename
-        """
-
-        # number of packets is the size of the filename / character size
-        num_packets = int(stat(filename)[6] / c_size)
-
-        # open the file
-        with open(filename, "rb") as f:
-            # for each packet
-            for i in range(num_packets + 1):
-                # move the cursor to the end of i * character size,
-                # add to buffer
-                f.seek(i * c_size)
-                f.readinto(send_buffer)
-
-                # return bytes; yield keyword returns without destroying
-                # states of local vars
-                yield bytes([i, 0x45, num_packets])
-
-    def save(self, dataset, savefile=None):
-        """
-        save the passed dataset to the passed savefile
-        dataset should be a set of lists; each line is a list:
-            save(([line1],[line2]))
-        to save a string, make it an item in a list:
-            save(['This is my string'])
-        by default, savefile is not passed
-        """
-        # if no savefile is passed, use the current filename attribute
-        # by default
-        if savefile is None:
-            savefile = self.filename
-
-        # open save file
+
+# initialize Satellite as cubesat
+cubesat = Satellite()
+
+"""
+IMU-related functions
+TODO: cubesat.imu hardware check
+"""
+
+def acceleration():
+    """ return the accelerometer reading from the IMU """
+    return cubesat.IMU.accel
+
+def magnetic():
+    """ return the magnetometer reading from the IMU """
+    return cubesat.IMU.mag
+
+def gyro():
+    """ return the gyroscope reading from the IMU """
+    return cubesat.IMU.gyro
+
+def temperature():
+    """ return the thermometer reading from the IMU """
+    return cubesat.IMU.temperature  # Celsius
+
+
+"""
+Burnwire-related functions
+TODO: cubesat.burnwire1 hardware check
+"""
+
+def burn(burn_num='1', dutycycle=0, freq=1000, duration=1):
+    """
+    control the burnwire(s)
+    initialize with burn_num = '1' ; burnwire 2 IC is not set up
+    """
+    # BURN1 = -Z,BURN2 = extra burnwire pin, dutycycle ~0.13%
+    dtycycl = int((dutycycle / 100) * (0xFFFF))
+
+    # print configuration information
+    print('----- BURN WIRE CONFIGURATION -----')
+    print(f'\tFrequency of: {freq}Hz')
+    print(f'\tDuty cycle of: {100 * dtycycl / 0xFFFF}% (int:{dtycycl})')
+    print(f'\tDuration of {duration}sec')
+
+    # initialize burnwire based on the burn_num passed to the function
+    if '1' in burn_num:
+        burnwire = cubesat.burnwire1
+    elif '2' in burn_num:
+        return False  # return False because burnwire 2 IC is not set up
```

Comment:
Should this throw an error?

Reply:
Ideally, yeah - I'm implementing that with hardware checks / want to put it in the next PR
Right now, it return "False" which indicates that deployment failed (but this case is not handled)

Reply:
ok